### PR TITLE
Flux VS Code extension

### LINKED ISSUE #678: Document Flux VS Code plugin
We have a Flux VS Code plugin!

##### Relevant URLs
https://marketplace.visualstudio.com/items?itemName=influxdata.flux

### LINKED ISSUE #1466: Tools: Visual Studio Code Flux extension - Update instructions and usage
The VS Code Flux language extension could use some clarifications in the following areas:
- Configuration steps for use with InfluxDB 1.x and 2.0
- Instructions on how to "Run Query" (it's somewhat context menu)
- Upgrading the extension

###### URL for relevant page?
https://marketplace.visualstudio.com/items?itemName=influxdata.flux
https://github.com/influxdata/vsflux

###### What products and version are you using?
InfluxDB 1.x and Cloud 2.0
Visual Studio Code v1.45.1
Flux extension v0.5.5

### LINKED ISSUE #1781: Tools: Visual Studio Code Flux language extension - No influxDB Server selected
Hi, 
I enabled flux-enabled = true in the .conf file and I'm successfully executing flux queries from Chronograf to influxdb v1.8.
When I try to execute the same queries from VS Code I receive this error "No influxDB server selected".
I tried to specify my local server address in the **from** (**host**: ...) as specificied in the docs (https://docs.influxdata.com/influxdb/v2.0/reference/flux/stdlib/built-in/inputs/from/) but it has no effect.
Can someone help me?

Example:
```
from(bucket: "tick_udemy/two_weeks/", host: "http://localhost:8086/")
  |> range(start: dashboardTime)
  |> filter(fn: (r) => r._measurement == "sensors" and (r._field == "temperature"))
  //|> window(every: 120m)
  |> mean()
  |> group(columns: ["_time", "_start", "_stop", "_value"], mode: "except")
```

Thanks.
Elisa

### PR DESCRIPTION
**Pull request by sanderson**
Closes #678
Closes #1466
Closes #1781

- Documents using the Flux VS Code extension with InfluxDB 1.8, 2.0, and Cloud
- Introduces new `{{< keybind >}}` shortcode for providing OS-specific keybindings/hotkeys

---

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/v2.0/tools/flux-vscode.md [hunk 1]
```diff
@@ -2,78 +2,80 @@
 title: Use the Flux VS Code extension
 seotitle: Use the Flux Visual Studio Code extension
 description: >
-  placeholder
+  The Flux VisuThe [Flux Visual Studio Code (VS Code) extension](https://marketplace.visualstudio.com/items?itemName=influxdata.flux)
```
Comment (mdevy-influxdata) [inline content/influxdb/v2.0/tools/flux-vscode.md:5]: Typo?

Comment (sanderson) [reply content/influxdb/v2.0/tools/flux-vscode.md:5]: Yes, thank you 😄

**Review by mdevy-influxdata — APPROVED** [review]

**Review by timhallinflux — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -333,6 +333,28 @@ WHERE time > now() - 15m
 {{< /code-tabs-wrapper >}}
 ~~~
 
+### Keybinds
+Use the `{{< keybind >}}` shortcode to include os-specific keybindings/hotkeys.
```
Comment (noramullen1) [inline CONTRIBUTING.md:337]: OS-specific?

File: content/influxdb/v1.8/tools/flux-vscode.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+- [Flux extension commands](#flux-extension-commands)
+
+## Install the Flux VS Code extension
+The Flux VS Code extension is available in the **Visual Studio Marketplace**.
+For information about installing extensions from the Visual Studio marketplace,
+see the [Extension Marketplace documentation](https://code.visualstudio.com/docs/editor/extension-gallery).
+
+## Connect to InfluxDB 1.8
+
+To create an InfluxDB Connection in VS Code:
```
Comment (noramullen1) [inline content/influxdb/v1.8/tools/flux-vscode.md:42]: lowercase C in connection

File: content/influxdb/v1.8/tools/flux-vscode.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+- [Upgrade the Flux extension](#upgrade-the-flux-extension)
+- [Flux extension commands](#flux-extension-commands)
+
+## Install the Flux VS Code extension
+The Flux VS Code extension is available in the **Visual Studio Marketplace**.
+For information about installing extensions from the Visual Studio marketplace,
+see the [Extension Marketplace documentation](https://code.visualstudio.com/docs/editor/extension-gallery).
+
+## Connect to InfluxDB 1.8
+
```
Comment (kelseiv) [inline content/influxdb/v1.8/tools/flux-vscode.md:41]: space after all?

File: content/influxdb/v1.8/tools/flux-vscode.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+## Query InfluxDB from VS Code
+1. Write your Flux query in a new VS Code file.
+2. Save your Flux script with the `.flux` extension or set the
+   [VS Code Language Mode](https://code.visualstudio.com/docs/languages/overview#_changing-the-language-for-the-selected-file) to **Flux**.
+3. Execute the query with the `influxdb.runQuery` command or {{< keybind mac="⌃⌥E" other="Ctrl+Alt+E" >}}.
+4. Query results appear in a new tab. If query results do not appear, see [Debug Flux queries](#debug-flux-queries).
+
+## Explore your schema
+With an InfluxDB connection configured, VS Code provides an overview of buckets,
```
Comment (noramullen1) [inline content/influxdb/v1.8/tools/flux-vscode.md:61]: After you've configured an InfluxDB connection,

File: content/influxdb/v1.8/tools/flux-vscode.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+measurements and tags in your InfluxDB organization.
```
Comment (kelseiv) [inline content/influxdb/v1.8/tools/flux-vscode.md:62]: oxford :-)

File: content/influxdb/v1.8/tools/flux-vscode.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+{{< img-hd src="/img/influxdb/1-8-tools-vsflux-explore-schema.png" alt="Explore your InfluxDB schema in VS Code" />}}
+
+## Debug Flux queries
+To view errors returned from Flux script executions, click the **Errors and Warnings**
+icons in the bottom left of your VS Code window, and then select the **Output** tab in the debugging pane.
+
+{{< img-hd src="/img/influxdb/2-0-tools-vsflux-errors-warnings.png" alt="VS Code errors and warnings"/>}}
+
+## Upgrade the Flux extension
+VS Code auto-updates by default extensions, but you are able to disable auto-update.
```
Comment (noramullen1) [inline content/influxdb/v1.8/tools/flux-vscode.md]: auto-updates extension by default, but you can disable it.

File: content/influxdb/v1.8/tools/flux-vscode.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+If auto-update is disabled, [manually update your VS Code Flux extension](https://code.visualstudio.com/docs/editor/extension-gallery#_update-an-extension-manually).
```
Comment (noramullen1) [inline content/influxdb/v1.8/tools/flux-vscode.md:75]: If you disable auto-update?

File: content/influxdb/v1.8/tools/flux-vscode.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+1. Write your Flux query in a new VS Code file.
+2. Save your Flux script with the `.flux` extension or set the
+   [VS Code Language Mode](https://code.visualstudio.com/docs/languages/overview#_changing-the-language-for-the-selected-file) to **Flux**.
+3. Execute the query with the `influxdb.runQuery` command or {{< keybind mac="⌃⌥E" other="Ctrl+Alt+E" >}}.
+4. Query results appear in a new tab. If query results do not appear, see [Debug Flux queries](#debug-flux-queries).
+
+## Explore your schema
+With an InfluxDB connection configured, VS Code provides an overview of buckets,
+measurements and tags in your InfluxDB organization.
+Use the **InfluxDB** pane in VS code to explore your schema.
```
Comment (kelseiv) [inline content/influxdb/v1.8/tools/flux-vscode.md:63]: how to open pane?

File: content/influxdb/v2.0/tools/flux-vscode.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
+---
+title: Use the Flux VS Code extension
```
Comment (noramullen1) [inline content/influxdb/v2.0/tools/flux-vscode.md:2]: same comments as 1.8 version

File: content/influxdb/v1.8/tools/flux-vscode.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+{{< img-hd src="/img/influxdb/1-8-tools-vsflux-explore-schema.png" alt="Explore your InfluxDB schema in VS Code" />}}
+
+## Debug Flux queries
+To view errors returned from Flux script executions, click the **Errors and Warnings**
+icons in the bottom left of your VS Code window, and then select the **Output** tab in the debugging pane.
```
Comment (kelseiv) [inline content/influxdb/v1.8/tools/flux-vscode.md:69]: icon

**Review by noramullen1 — APPROVED** [review]

File: content/influxdb/v1.8/tools/flux-vscode.md (hunk 7 above)
Comment (kelseiv) [inline content/influxdb/v1.8/tools/flux-vscode.md]: by default extensions?

File: content/influxdb/v1.8/tools/flux-vscode.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+
+{{< img-hd src="/img/influxdb/2-0-tools-vsflux-errors-warnings.png" alt="VS Code errors and warnings"/>}}
+
+## Upgrade the Flux extension
+VS Code auto-updates by default extensions, but you are able to disable auto-update.
+If auto-update is disabled, [manually update your VS Code Flux extension](https://code.visualstudio.com/docs/editor/extension-gallery#_update-an-extension-manually).
+After updating the extension, reload VS Code to initialize the updated extensions.
```
Comment (kelseiv) [inline content/influxdb/v1.8/tools/flux-vscode.md:76]: how to reload?

File: content/influxdb/v1.8/tools/flux-vscode.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+
+## Flux extension commands
+
+| Command                     | Description       | Keyboard shortcut                            | Menu context      |
```
Comment (kelseiv) [inline content/influxdb/v1.8/tools/flux-vscode.md:80]: is menu context the only menus where command is available?

**Review by kelseiv — APPROVED** [review]

File: content/influxdb/v1.8/tools/flux-vscode.md (hunk 11 above)
Comment (sanderson) [reply content/influxdb/v1.8/tools/flux-vscode.md:69]: There are two icons.

File: content/influxdb/v1.8/tools/flux-vscode.md (hunk 9 above)
Comment (sanderson) [reply content/influxdb/v1.8/tools/flux-vscode.md:63]: There's an image below that shows the pane.

File: content/influxdb/v1.8/tools/flux-vscode.md (hunk 13 above)
Comment (sanderson) [reply content/influxdb/v1.8/tools/flux-vscode.md:81]: Yes, that's correct.

**Pull request merged**